lib/transport.dart

```diff
@@ -210,6 +210,11 @@ abstract class TransportStream {
   /// A sink for writing data and/or headers to the remote end.
   StreamSink<StreamMessage> get outgoingMessages;
 
+  /// Set the cancellation handler on this stream.
+  ///
+  /// The handler will be called if the stream receives an RST_STREAM frame.
+  void onCancel(void value(int errorCode));
```

Comment:
I'll rename it to onTerminated and make it a setter.

I think it's better to have both. The termination handler is an extra thing that's useful in some cases like gRPC, where we might be done reading inputs, and busy doing work before sending a response, but would still like to know if the stream has been terminated.